ext/droption/droption.h

```diff
@@ -710,25 +716,50 @@ template <>
 inline bool
 droption_t<unsigned long>::convert_from_string(const std::string s)
 {
-    errno = 0;
-    long input = strtol(s.c_str(), NULL, 10);
-    if (input >= 0)
-        value_ = (unsigned long)input;
-    else
+    // Checks if the first non-space character of a string is a negative sign. If it is,
+    // it is invalid.
+    const char *str = s.c_str();
+    const char *startptr = &str[0];
+    for (int i = 0; i < (int)s.size(); i++) {
+        char c = str[i];
+        if (c == ' ' || c == '\n' || c == '\r' || c == '\t' || c == '\v' || c == '\f') {
+            continue;
+        } else {
+            startptr = &str[i];
+            break;
+        }
+    }
+    if (*startptr == '-') {
```

Comment:
@derekbruening
We finally find that abseil checks whether the input is a negative digital string before parsing the unsigned int/long/long long. And abseil uses this [function](https://github.com/abseil/abseil-cpp/blob/736458b555768b8d8d97883eab57ad54c5c9fc1a/absl/strings/ascii.h#L96) to check whether the character is a space. In our parsing function, we do the same thing: skip all spaces and check if the first non-space is a negative sign. Do you agree with the current solution?

Reply:
That whitespace check does not look good enough to me: that is only for ASCII and will fail on non-ASCII whitespace.  Core DR also has an ASCII-only isspace but at least it has something of an excuse where it can't safely use standard libraries.  Is there a problem with calling the libc `isspace`?  Frontends/standalones should be ok with that, and clients should be ok with the private loader in place.

Reply:
```C
for (size_t i = 0; i < s.size(); i++) {
  if (isspace(s[i])) continue;
  if (s[i] == '-') return false;
}
```

Reply:
Done in new PR #5552 .